src/Workspaces/Core/Portable/SymbolKey/SymbolKey.BodyLevelSymbolKey.cs

```diff
@@ -132,20 +132,22 @@ public static SymbolKeyResolution Resolve(SymbolKeyReader reader, out string? fa
                 }
 
                 // First check if we can recover the symbol just through the original location.
-                foreach (var loc in locations)
+
+                string? totalFailureReason = null;
+                for (var i = 0; i < locations.Count; i++)
                 {
-                    var resolutionOpt = reader.ResolveLocation(loc);
-                    if (resolutionOpt.HasValue)
+                    var loc = locations[i];
+
+                    if (!TryResolveLocation(loc, i, out var resolution, out var reason))
                     {
-                        var resolution = resolutionOpt.Value;
-                        var symbol = resolution.GetAnySymbol();
-                        if (symbol?.Kind == kind &&
-                            SymbolKey.Equals(reader.Compilation, name, symbol.Name))
-                        {
-                            failureReason = null;
-                            return resolution;
-                        }
+                        totalFailureReason = totalFailureReason == null
+                            ? $"({reason})"
+                            : $"({totalFailureReason} -> {reason})";
```

Comment:
This is going to add up to a string like (((a) -> b) -> c) with a bunch of nested parenthesis; expected?

Reply:
yup. that way you can make out the 'tree/nesting' structure.  i.e. you might have a few failures, which themselves have nested failures.  i.e.:

`(location 1 failed because (foo failed bacause (bar failed ...))) (location 2 failed because (foo failed because (...))))`

not havin thte parens makes it ambiguous how this ties together.